src/Instrumentation/ExtAmqp/tests/Integration/ExtAmqpInstrumentationTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Instrumentation\ExtAmqp\tests\Integration;
+
+use AMQPQueue;
+use ArrayObject;
+use OpenTelemetry\API\Instrumentation\Configurator;
+use OpenTelemetry\API\Trace\Propagation\TraceContextPropagator;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\Context\ScopeInterface;
+use OpenTelemetry\SDK\Trace\ImmutableSpan;
+use OpenTelemetry\SDK\Trace\SpanExporter\InMemoryExporter;
+use OpenTelemetry\SDK\Trace\SpanProcessor\SimpleSpanProcessor;
+use OpenTelemetry\SDK\Trace\TracerProvider;
+use OpenTelemetry\SemConv\TraceAttributes;
+use PHPUnit\Framework\TestCase;
+
+class ExtAmqpInstrumentationTest extends TestCase
+{
+    private ScopeInterface $scope;
+    private ArrayObject $storage;
+
+    public function setUp(): void
+    {
+        $this->storage = new ArrayObject();
+        $tracerProvider = new TracerProvider(
+            new SimpleSpanProcessor(
+                new InMemoryExporter($this->storage)
+            )
+        );
+
+        $this->scope = Configurator::create()
+            ->withTracerProvider($tracerProvider)
+            ->withPropagator(TraceContextPropagator::getInstance())
+            ->activate();
+    }
+
+    public function tearDown(): void
+    {
+        $this->scope->detach();
+    }
+
+    public function test_rabbit_basic_publish_without_args_works(): void
+    {
+        list($connection, $routing_key, $channel, $exchange, $queue) = $this->setUpQueue();
+
+        try {
+            $exchange->publish('test', $routing_key);
+
+            $this->assertCount(1, $this->storage);
+
+            /** @var ImmutableSpan $span */
+            $span = $this->storage[0];
+
+            $this->assertEquals($routing_key . ' publish', $span->getName());
+            $this->assertEquals('amqp', $span->getAttributes()->get(TraceAttributes::MESSAGING_SYSTEM));
+            $this->assertEquals(SpanKind::KIND_PRODUCER, $span->getKind());
+
+            /**
+             * Our message should be the first one in the queue
+             */
+            $envelope = $queue->get();
+
+            self::assertFalse($envelope->hasHeader('traceparent'), 'traceparent header is present');
```

Comment:
So the way I'm used to using PHPUnit messages is: describe the condition that is unexpected. So in this case, if the test fails, it spells out what is unexpected.

 Should I switch both to the other way around?

Reply:
> describe the condition that is unexpected

I think it's more of an art than a science, and from skimming the phpunit docs I didn't find any guidance. I've always thought of the message as a human-readable form of the expectation, but there are plenty of examples of the message describing the failure (which makes sense too, since it's only displayed in case of failure).
Let's leave it as-is :)